### PR TITLE
Add incarnation to acc.root in IH

### PR DESCRIPTION
**Pull request by AskAlexSharov**

### CODE REVIEW

File: core/rawdb/accessors_account.go [hunk 1]
```diff
@@ -24,44 +24,64 @@ import (
...
 	}
-	if err := db.Put(dbutils.IntermediateTrieHashBucket, addrHashBytes, acc.Root.Bytes()); err != nil {
+	if err := db.Put(dbutils.IntermediateTrieHashBucket, dbutils.GenerateStoragePrefix(addrHash, acc.Incarnation), acc.Root.Bytes()); err != nil {
 		return err
 	}
 	return nil
 }
 
-func DeleteAccount(db DatabaseDeleter, addrHash common.Hash) error {
+type DBToDeleteAccount interface {
```
Comment (mandrigin) [inline core/rawdb/accessors_account.go:62]: Maybe just `DatabaseReaderDeleter`?

Comment (AskAlexSharov) [reply core/rawdb/accessors_account.go:62]: right

File: trie/resolver_stateful_cached.go [hunk 2]
```diff
@@ -251,7 +251,20 @@ func (tr *ResolverStatefulCached) Walker(isAccount bool, blockNr uint64, fromCac
 		tr.curr.Write(tr.succ.Bytes())
 		tr.succ.Reset()
 		skip := tr.currentReq.extResolvePos // how many first nibbles to skip
-		tr.succ.Write(kAsNibbles[skip:])
+
+		//tr.succ.Write(kAsNibbles[skip:])
```
Comment (mandrigin) [inline trie/resolver_stateful_cached.go:255]: 1) the comment seems to be a leftover

2) we have `type hexWriter struct {` in `core/types/derive_sha.go`. We might want to make it public and just reuse it here.

Comment (AskAlexSharov) [reply trie/resolver_stateful_cached.go:255]: `hexWriter` - can't because here we may need to write even amount of nibbles.

**Pull request merged**